src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part5_update_records.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+import argparse
+from collections import Counter
+import gzip
+import pysam
+import sys
+import svtk.utils as svu
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('revise_vcf_lines', type=argparse.FileType('r'))
+    parser.add_argument('normal_revise_vcf')
+    parser.add_argument('famfile', type=argparse.FileType('r'))
+    parser.add_argument('sexchr_revise')
+    parser.add_argument('multi_geno_ids_txt')
+    parser.add_argument('outlier_samples_list', type=argparse.FileType('r'))
+    args = parser.parse_args()
+
+    # load the revised lines and index by ID
+    revised_lines_by_id = {}
+    with pysam.VariantFile(args.revise_vcf_lines) as revise_vcf:
+        header2 = revise_vcf.header
+        for record in revise_vcf:
+            revised_lines_by_id[record.id] = record
+    print("loaded {} revised lines".format(len(revised_lines_by_id)), file=sys.stderr)
+
+    outlier_samples = set([line.rstrip() for line in args.outlier_samples_list if not line.isspace()])
+    print("loaded {} outlier samples".format(len(outlier_samples)), file=sys.stderr)
+
+    male_samples = set()
+    for line in args.famfile:
+        if line.isspace():
+            continue
+        fields = line.rstrip().split("\t")
+        if fields[4] == '1':
+            male_samples.add(fields[1])
+    print("identified {} male samples".format(len(male_samples)), file=sys.stderr)
+
+    if args.sexchr_revise.endswith(".gz"):
+        sexchr_revise = set([line.rstrip() for line in gzip.open(args.sexchr_revise, 'rt')])
+    else:
+        sexchr_revise = set([line.rstrip() for line in open(args.sexchr_revise, 'rt')])
+    print("{} sites to revise on sex chromosomes".format(len(sexchr_revise)), file=sys.stderr)
+
+    if args.multi_geno_ids_txt.endswith(".gz"):
+        multi_geno_ids = set([line.rstrip() for line in gzip.open(args.multi_geno_ids_txt, 'rt')])
+    else:
+        multi_geno_ids = set([line.rstrip() for line in open(args.multi_geno_ids_txt, 'rt')])
+    print("{} multiallelic sites".format(len(multi_geno_ids)), file=sys.stderr)
+
+    normal_vcf = pysam.VariantFile(args.normal_revise_vcf)
```

Comment:
It would be best practices to use a context manager (like you do at the top of the file). Since all the files you open are known at this point, you can do them all at once and only have one indent level:
```
with open("foo", 'r') as foo_in, pysam.VariantFile("bar.vcf", 'w') as f_out, pysam.VariantFile("snarf", 'r') as snarf_in:
    do_stuff()
```

Reply:
I couldn't make this work for all the input and output files at once since the outputs need the modified header from the input, but I was able to use two nested `with` statements.